File: BusDepart/app/src/main/java/dk/itu/busdepart/HttpThread.java
package dk.itu.busdepart;
import android.widget.TextView;

public class HttpThread implements Runnable {
  // to pass parameters to the thread
  String url;
  TextView ui;

  public HttpThread(String url, TextView ui) {
    this.ui= ui;  this.url= url;
  }

  public void run() {
    String result= new NetworkFetcher().fetchItems(url);
    ui.post(() -> ui.append(result));

    //Why not    ui.append(result)
  }
}

File: HttpAssets/app/src/main/java/dk/itu/httpassets/HttpThread.kt
package dk.itu.httpassets
import android.widget.TextView
import java.lang.Runnable

class HttpThread(private var url: String, private var result: TextView) : Runnable {
  override fun run() {
    val assetString = NetworkFetcher().fetchItems(url)
    result.post { result.text = assetString }
  }
}